### PR TITLE
chore: add types for Cypress.session.clearAllSavedSessions

### LINKED ISSUE #19363: [typescript] Cypress.session type does not exist
### Current behavior

I tried to use `Cypress.session.clearAllSavedSessions();`, but got a typescript error.

image

### Desired behavior

`Cypress.session` should be typed correctly.

### Test code to reproduce

https://github.com/IanVS/cypress-test-tiny

`npm i`
`npm run typecheck`

### Cypress Version

9.1.1

### Other

_No response_

### PR DESCRIPTION
**Pull request by IanVS**
- Closes #19363 

### User facing changelog

n/a

### Additional details

I only added the public method that's documented, although there are other methods available.  

I'm not positive I did this right.  😬  And I didn't see any type tests to modify.  Is the return type of `Promise<void>` correct? 

### How has the user experience changed?

Typescript types for `Cypress.session` should now be available.

### PR Tasks

- [na] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: cli/types/cypress.d.ts [hunk 1]
```diff
@@ -3117,6 +3122,11 @@ declare namespace Cypress {
     onAnyAbort(route: RouteOptions, proxy: any): void
   }
 
+  interface Session {
+    clearAllSavedSessions: () => Promise<void>;
```
Comment (emilyrohrbough) [inline cli/types/cypress.d.ts:3126]: Mind adding a comment to describe clearAllSavedSessions? It helps with maintenance. You can pull something from https://docs.cypress.io/api/cypress-api/session.

Comment (IanVS) [reply cli/types/cypress.d.ts:3126]: Unfortunately there's no docs there either, but I can try to write something up.

Comment (emilyrohrbough) [reply cli/types/cypress.d.ts:3126]: It mentions something like;` Clear all saved sessions and re-run the current spec file.`

Comment (IanVS) [reply cli/types/cypress.d.ts:3126]: Yeah sorry, I missed that at first.  :shame:

Comment (emilyrohrbough) [reply cli/types/cypress.d.ts:3127]: No worries! It's not super obvious

File: cli/types/cypress.d.ts [hunk 2]
```diff
@@ -3117,6 +3122,12 @@ declare namespace Cypress {
     onAnyAbort(route: RouteOptions, proxy: any): void
   }
 
+  interface Session {
+    // Clear all saved sessions and re-run the current spec file.
+    clearAllSavedSessions: () => Promise<void>;
+
```
Comment (emilyrohrbough) [inline cli/types/cypress.d.ts:3128]:
```suggestion
```

File: cli/types/cypress.d.ts [hunk 3]
```diff
@@ -3117,6 +3122,11 @@ declare namespace Cypress {
     onAnyAbort(route: RouteOptions, proxy: any): void
   }
 
+  interface Session {
+    // Clear all saved sessions and re-run the current spec file.
+    clearAllSavedSessions: () => Promise<void>;
```
Comment (emilyrohrbough) [inline cli/types/cypress.d.ts:3127]:
```suggestion
    clearAllSavedSessions: () => Promise<void>
```

**Review by BlueWinds — APPROVED** [review]

**Review by emilyrohrbough — APPROVED** [review]

**Pull request merged**